SentryTestUtils/SentryProfilerMocksSwiftCompatible.h

```diff
@@ -0,0 +1,28 @@
+#import "SentryProfilingConditionals.h"
+#import <Foundation/Foundation.h>
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+@class SentryProfilerState;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * This delivers a wrapper around the C++ function to create a mock backtrace for incorporation into
+ * profiler state that can be called from Swift tests.
+ */
+@interface SentryProfilerMocksSwiftCompatible : NSObject
```

Comment:
`l`: Mocks are usually test doubles that simulate the behavior of a real object with the added ability to verify interactions with that object. They are used to set expectations and assert that specific methods or properties of the mocked object are called or accessed correctly during testing. What you define here is a fake. Fakes are simplified implementations of a real object that mimics its behavior to some extent. They are typically used to provide a functional substitute for a real object that is too complex, slow, or unavailable for testing.

TLDR; if you want to follow the correct testing naming terms, we should rename all the mocks in this PR to fake, but I'm also fine with keeping it. I don't want to be nitpicking on testing terms.

Reply:
Sorry but I think this is more of a mock. I think we should be careful introducing too much terminology (and therefore cognitive overhead) like fixture, double, oracle, stub, mock, fake, dummy, spy, robot... I don't think these are all standardized terms in industry. 

Since this is just creating structs to use as input to a real implementation, and not actually providing an alternate implementation of an interface, there is really no doubling going on at all, so in an OOP sense, this is neither mocking nor stubbing (nor faking).

I think the intention of the code gets across so I'm hesitant to change it. We're providing predefined input to the SUT and testing the output of the system.

Reply:
Fine by me 👍 .